### ISSUE TITLE
[Request] Training/Testing code and class weighting scheme for WCE loss

### ISSUE THREAD

**Issue by somatdu** [jie666-6/UrbanSARFloods#6]
### Summary
Thank you very much for releasing the **UrbanSARFloods dataset** and the CVPRW 2024 paper :contentReference[oaicite:1]{index=1}.  
I am currently working on reproducing your experiments and would like to kindly ask for more details to ensure reproducibility.

---

### Requests
1. **Training / Testing code**
   - Could you please share the program code (training, evaluation, testing scripts, configs, requirements) used in the experiments?
   - Even a minimal training recipe would be very helpful.

2. **Weighted Cross-Entropy (WCE) loss**
   - In the paper, you mention that WCE was employed to handle class imbalance :contentReference[oaicite:2]{index=2}.
   - Could you please provide the **exact class weights** (for NF, FO, FU) used in your experiments?
   - Additionally, could you clarify **how the weights were computed** (e.g., inverse frequency, normalization method, which split was used, whether ignore_index was applied, etc.)?

---

### Background
- I am attempting to replicate the baseline experiments (U-Net, U-Net++, etc.) using Segmentation Models PyTorch as stated in the paper.  
- Since the dataset exhibits extreme imbalance (especially for FU), having the exact weighting scheme is crucial to reproduce the results.  
- For reference, my preliminary counts from training masks were:  
  `counts = [NF, FO, FU] = [1161425968, 18120626, 363550]`  
  I would like to confirm whether your calculation and resulting weights are consistent with this.

---

### Expected Output
- Training/testing code (or link to repository/branch if available).  
- The **numerical values of the WCE class weights** and a brief explanation of the calculation procedure.

Thank you in advance for your kind support.  
Your help will greatly contribute to reproducibility and further research using UrbanSARFloods.

**Comment by jie666-6** [comment]
The models were implemented using the open-source segmentation_models_pytorch (SMP) library.
The training follows a standard segmentation pipeline (e.g., binary cross-entropy loss, Adam optimizer) as described in the paper.
Therefore, we did not release separate training scripts, since all procedures can be easily reproduced with SMP.

For the weights calculation
num_0  = 1161425968
num_1 =  18120626
num_2 = 363550
weight_in = [num_0 / num_0, num_0/ num_1, num_0 / num_2 ]
weight_in = torch.tensor(weight_in,device=device)
